jablib/src/main/java/org/jabref/logic/exporter/AcademicPagesExporter.java

```diff
@@ -0,0 +1,91 @@
+ */
+public class AcademicPagesExporter extends Exporter {
+    private static final String BLANK_LINE_PATTERN = "\\r\\n|\\n";
+    private static final String LAYOUT_PREFIX = "/resource/layout/";
+    private static final String LAYOUT_EXTENSION = ".layout";
+    private static final String FORMATTERS_EXTENSION = ".formatters";
+    private static final String BEGIN_INFIX = ".begin";
+    private static final String END_INFIX = ".end";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TemplateExporter.class);
+
+    private final String lfFileName;
+    private final String directory;
+    private final LayoutFormatterPreferences layoutPreferences;
+    private final SelfContainedSaveOrder saveOrder;
+    private boolean customExport;
+    private List<BibEntry> entries;
+    private final TemplateExporter academicPagesTemplate;
+
+    /**
+     * Initialize another export format based on templates stored in dir with layoutFile lfFilename.
+     *
+     */
+    public AcademicPagesExporter(LayoutFormatterPreferences layoutPreferences, SelfContainedSaveOrder saveOrder) {
+        super("academicpages", "academicpages", StandardFileType.MARKDOWN);
+        this.lfFileName = "academicpages";
+        this.directory = "academicpages";
+        this.layoutPreferences = layoutPreferences;
+        this.saveOrder = saveOrder;
+        String consoleName = "academicpages";
+        this.academicPagesTemplate = new TemplateExporter("academicpages", consoleName, lfFileName, directory, StandardFileType.MARKDOWN, layoutPreferences, saveOrder);
+    }
+
+    @Override
+    public void export(@NonNull final BibDatabaseContext databaseContext,
+                       final Path exportDirectory,
+                       @NonNull List<BibEntry> entries) throws SaveException {
+        export(databaseContext, exportDirectory, entries, List.of(), JournalAbbreviationLoader.loadBuiltInRepository());
+    }
+
+    /**
+     * The method that performs the export of all entries by iterating on the entries.
+     *
+     * @param databaseContext the database to export from
+     * @param exportDirectory            the directory to write to
+     * @param entries         a list containing all entries that should be exported
+     * @param abbreviationRepository the built-in repository
+     * @throws SaveException   Exception thrown if saving goes wrong
+     */
+    @Override
+    public void export(@NonNull final BibDatabaseContext databaseContext,
+                       final Path exportDirectory,
+                       @NonNull List<BibEntry> entries,
+                       List<Path> fileDirForDataBase,
+                       JournalAbbreviationRepository abbreviationRepository) throws SaveException {
+        if (entries.isEmpty()) { // Only export if entries exist
+            return;
+        }
+        try {
+            Integer iterator = 1;
+            for (BibEntry entry : entries) {
```

Comment:
You could simply switch this to a normal for loop  with for (int i =1; i< entiries.length; i++) so you don't need the iterator variable.